core/src/HIP/Kokkos_HIP.cpp

```diff
@@ -22,6 +22,18 @@ import kokkos.core;
 
 #include <iostream>
 
+namespace {
+
+struct CustomDeleterManagesStream {
+  void operator()(Kokkos::Impl::HIPInternal* ptr) const {
+    hipStream_t stream = ptr->m_stream;
+    delete ptr;
+    KOKKOS_IMPL_HIP_SAFE_CALL(hipStreamDestroy(stream));
+  }
+} customDeleterManagesStream;
```

Comment:
Why do you need to name the struct?
```suggestion
struct {
  void operator()(Kokkos::Impl::HIPInternal* ptr) const {
    hipStream_t stream = ptr->m_stream;
    delete ptr;
    KOKKOS_IMPL_HIP_SAFE_CALL(hipStreamDestroy(stream));
  }
} customDeleterManagesStream;
```
If you there is a reason to have a name, I find choosing the same name as for the variable very confusing.

Reply:
OK will apply next time I push

Reply:
I don't care too much but I wonder if the debug info is better when you look at the symbols with a named structure VS anonymous